src/lib/utils/hooks/useSellerBlacklist.ts

```diff
@@ -0,0 +1,96 @@
+import { accounts, subgraph } from "@bosonprotocol/react-kit";
+import { useMemo } from "react";
+import { useQuery } from "react-query";
+
+import { CONFIG } from "../../config";
+import { fetchTextFile } from "../textFile";
+import { useCurrentSellers } from "./useCurrentSellers";
+
+interface Props {
+  sellerBlacklistUrl?: string;
+  allowConnectedSeller?: boolean;
+}
+
+export function useSellerBlacklist(
+  props: Props,
+  options: { enabled: boolean } = { enabled: true }
+): {
+  isSuccess: boolean;
+  isError: boolean;
+  curatedSellerIds: string[] | undefined;
+} {
+  const currentSeller = useCurrentSellers();
+  const allSellers = useQuery(
+    ["all-sellers", props],
+    async () => {
+      const maxSellerPerReq = 1000;
+      const fetched = await accounts.subgraph.getSellers(CONFIG.subgraphUrl, {
+        sellersFirst: maxSellerPerReq,
+        sellersOrderBy: subgraph.Seller_OrderBy.SellerId,
+        sellersOrderDirection: subgraph.OrderDirection.Asc
+      });
+      let loop = fetched.length === maxSellerPerReq;
+      let sellersSkip = maxSellerPerReq;
+      while (loop) {
+        const toAdd = await accounts.subgraph.getSellers(CONFIG.subgraphUrl, {
+          sellersFirst: maxSellerPerReq,
+          sellersSkip,
+          sellersOrderBy: subgraph.Seller_OrderBy.SellerId,
+          sellersOrderDirection: subgraph.OrderDirection.Asc
+        });
+        fetched.push(...toAdd);
+        loop = toAdd.length === maxSellerPerReq;
+        sellersSkip += maxSellerPerReq;
+      }
+      return fetched;
+    },
+    {
+      ...options
+    }
+  );
+  const blacklist = useQuery(
+    "seller-blacklist",
+    async () => {
+      const blacklistStr = props.sellerBlacklistUrl
+        ? await fetchTextFile(props.sellerBlacklistUrl, false)
```

Comment:
should we catch this exception if this fails and then throw it so we can capture it with Sentry and alert us if the json file we use for blacklisting cannot be fetched?

Reply:
Ok, to capture the error with Sentry. Will do it in fetchTextFile below.

Reply:
Here my intention is to let the exception throwing to ensure `useQuery()` is failing (retrying N times?, then returning `isError=true`)